src/main/scala/beam/utils/ScenarioReaderCsv.scala

```diff
@@ -156,7 +159,12 @@ object ScenarioReaderCsv {
         population.getPersonAttributes.putAttribute(person.getId.toString, "householdId", household_id)
         population.getPersonAttributes.putAttribute(person.getId.toString, "rank", 0)
         population.getPersonAttributes.putAttribute(person.getId.toString, "age", age.toInt)
-        population.getPersonAttributes.putAttribute(person.getId.toString, "available-modes", modes)
+        val attributesOfIndividual = PopulationAdjustment
+          .adjustBeamAttributes(population, person.getId.toString, modes.split(","))
+        population.getPersons
```

Comment:
can this additional operation be hidden from caller of module?

Reply:
Did you mean to club both the steps (fetching attributesOfIndividual & setting it inside beam-attributes) into a single function @wrashid ?